### ISSUE TITLE
Panic when verifying malformed signed cookie

### ISSUE THREAD

**Issue by 5225225** [SergioBenitez/cookie-rs#178]
reproduction:

```rust
fn main() {
    let data = "x=yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy£";

    let c = cookie::Cookie::parse(data).expect("failed to parse cookie");

    let key = cookie::Key::from(&[0u8; 64]);

    let mut jar = cookie::CookieJar::new();

    let signed = jar.signed_mut(&key);

    signed.verify(c).expect("failed to verify cookie");
}
```

stack trace:

```
thread 'main' panicked at 'byte index 44 is not a char boundary; it is inside '£' (bytes 43..45) of `yyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyyy£`', /rustc/952fdf2a1119affa1b37bcacb0c49cf9f0168ac8/library/core/src/str/mod.rs:576:13
stack backtrace:
   0: rust_begin_unwind
             at /rustc/952fdf2a1119affa1b37bcacb0c49cf9f0168ac8/library/std/src/panicking.rs:515:5
   1: core::panicking::panic_fmt
             at /rustc/952fdf2a1119affa1b37bcacb0c49cf9f0168ac8/library/core/src/panicking.rs:92:14
   2: core::str::slice_error_fail
   3: core::str::<impl str>::split_at
             at /rustc/952fdf2a1119affa1b37bcacb0c49cf9f0168ac8/library/core/src/str/mod.rs:576:13
   4: cookie::secure::signed::SignedJar<J>::_verify
             at /home/jess/.cargo/git/checkouts/cookie-rs-9c795d75da5c954b/0de451d/src/secure/signed.rs:57:35
   5: cookie::secure::signed::SignedJar<J>::verify
             at /home/jess/.cargo/git/checkouts/cookie-rs-9c795d75da5c954b/0de451d/src/secure/signed.rs:94:28
   6: scratchQVE5xNE9W::main
             at ./src/main.rs:12:5
   7: core::ops::function::FnOnce::call_once
             at /rustc/952fdf2a1119affa1b37bcacb0c49cf9f0168ac8/library/core/src/ops/function.rs:227:5
```

Tested on 0de451d9defefd0018a00b6001d2671ac2e33fdc as well as 0.15.0 from crates.io

**Comment by SergioBenitez** [comment]
Turns out `split_at` for a string and `split_at` for a `Vec` panic for different reasons. Bummer. Pushing a fix immediately.

**Comment by SergioBenitez** [comment]
Published in 0.15.1.
